getAddresses.cpp

```diff
@@ -0,0 +1,53 @@
+// Takes pID of the process as input and returns the addresses of the first element of 2d array. 
+// Also stroes the number of rows in AddressesRows and address of the first elemet in AddressesList,
+
+#include "ProcessManager.h"
+using namespace std;
+int** ProcessManager :: getAddresses (long unsigned int pID) {
+	int fd = 0;
+	char FileLocation[1024];
+	char ChBaseAddress[1024];
+	char* p = NULL;
+	sprintf(FileLocation,"/proc/%lu/maps",pID);
+	FILE *ptr;
+	if ( (ptr = fopen(FileLocation, "r")) == NULL  ) {
+		printf( " Error in opening file");
+		exit (1);
+	}
+	ptr = fopen (FileLocation, "r");
+	char line [1000];
+	int i =0;
+	int count =0;
+	
+	int** AddressesList = (int**)malloc(30*sizeof(int*));
+	for(int i =0;i<30;i++) {
+		AddressesList[i] = new int[2];
+	}
```

Comment:
why are you allocating only 30 pointers?
Maps file can give you very high number of concurrent memory locations (definitely more than 30).
If you are doing something else, please explain.

Reply:
oh, the maps files I've seen didn't have more rows than that, so I kept that, I'll increase the number.

Reply:
Limiting this would be not a great way because most of the maps file we are seeing right now are those of very simple programs.
For more info every malloc can potentially create a separate entry in maps file.
Now imagine a very big program and how many mallocs it might be performing under the hood.
This is why in my opinion limiting this number will not be a good idea.

Reply:
I see...what would you suggest then?

Reply:
Sure. How you want to solve the problem is your choice. I am here just to review and guide.